### PR TITLE
adding a while loop

### PR DESCRIPTION
**Pull request by JonathanBanerjee**
@lgiordani

What does this PR do?

Adding the while loop to pg_isready

How should this be manually tested?
By executing the bash script in the command line.

Any background context you want to provide?
N/A

External context (Link to ticket etc)
N/A

### CODE REVIEW

File: create_db.sh [hunk 1]
```diff
@@ -15,18 +15,20 @@ docker volume rm wordle_dbdata
 echo "Now running the postgres container"
 container_id=$(docker run -it -d  -e POSTGRES_USER=${pg_user} -e POSTGRES_DB=${pg_db} -e POSTGRES_PASSWORD=${pg_pass} -v $PWD:/opt/sql -v wordle_dbdata:/var/lib/postgresql/data postgres:13) 
 
-
+while ! docker exec ${container_id} pg_isready; do
```
Comment (lgiordani) [inline create_db.sh]: I recommend writing tests always with the full syntax `[[ test ]]`. Bash has loads of corner cases and it's better to be very strict about such things. Have you checked that the exit status of `docker exec` is the same of `pg_ready`? The rest of the loop is correct.

**Review by lgiordani — CHANGES_REQUESTED** [review]

File: create_db.sh [hunk 2]
```diff
@@ -11,16 +11,27 @@ if [[ $(docker volume ls -f name=wordle_dbdata) == "wordle_dbdata" ]]; then dock
...
+  fi
   # Print a message and sleep for 3 seconds before checking again
-  echo "Waiting for the database to start"
-  sleep 1
+  echo "Waiting for the database to start..."
+  sleep 3
 done
 
+# Check the exit status of the docker exec command
+if [[ $? -ne 0 ]]; then
```
Comment (lgiordani) [inline create_db.sh:40]: Can this even happen? I think if you reached this point it's because `$?` is 0. Anyway, no harm done.
Food for thoughts: the reason why Docker Compose and health checks are great is because they simplify all this work. For example, your code is waiting forever, so a possible upgrade would be to check N times and then to give up. But for now it looks good. :top:

**Review by lgiordani — APPROVED** [review]

File: create_db.sh [hunk 3]
```diff
@@ -21,6 +23,14 @@ while true; do
     # The server is ready, so exit the loop
     break
   fi
+  # Increment the counter variable
+  counter=$((counter + 1))
+
+  if [[ $counter -ge $loop_limit ]]; then
```
Comment (lgiordani) [inline create_db.sh]:
```suggestion
  if [[ ${counter} -ge ${loop_limit} ]]; then
```

**Review by lgiordani — APPROVED** [review]

File: create_db.sh (hunk 3 above)
Comment (JonathanBanerjee) [reply create_db.sh]: Hi @lgiordani , I have added this change in my next commit, is it ready to merge?
